syntax = "proto3";

package campaign;

option go_package = "campaign-svc/proto;campaign";

service CampaignService {
  rpc GetCampaign(GetCampaignRequest) returns (GetCampaignResponse);
  rpc CheckCampaignActive(CheckCampaignActiveRequest) returns (CheckCampaignActiveResponse);
}

message GetCampaignRequest {
  int32 campaign_id = 1;
}

message GetCampaignResponse {
  int32 id = 1;
  string title = 2;
  string name = 3;
  string description = 4;
  int32 product_id = 5;
  string start_date = 6;
  string end_date = 7;
  bool is_active = 8;
  string created_at = 9;
  string updated_at = 10;
}

message CheckCampaignActiveRequest {
  int32 campaign_id = 1;
}

message CheckCampaignActiveResponse {
  bool is_active = 1;
  string message = 2;
  string end_date = 3;
}
